devicetypes/smartthings/neo-coolcam-light-switch.src/neo-coolcam-light-switch.groovy

```diff
@@ -0,0 +1,164 @@
+
+		standardTile("refresh", "device.switch", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "default", label: '', action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+
+		main "switch"
+		details(["switch", "refresh"])
+	}
+}
+
+def installed() {
+	def componentLabel = "NEO Coolcam Light Switch ep2"
+	try {
+		String dni = "${device.deviceNetworkId}-ep2"
+		addChildDevice("Binary Switch Endpoint", dni, device.hub.id,
+			[completedSetup: true, label: "${componentLabel}",
+			 isComponent   : false, componentName: "ch2", componentLabel: "${componentLabel}"])
+		log.debug "Endpoint 2 (Binary Switch Endpoint) added as $componentLabel"
+	} catch (e) {
+		log.warn "Failed to add endpoint 2 ($desc) as Binary Switch Endpoint - $e"
+	}
+	configure()
+}
+
+def updated() {
+	configure()
+}
+
+def configure() {
+	// Device-Watch simply pings if no device events received for checkInterval duration of 32min = 2 * 15min + 2min lag time
+	sendEvent(name: "checkInterval", value: 30 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+	response([zwave.configurationV1.configurationSet(parameterNumber: 4, size: 1, configurationValue: [1]).format(),
+			  "delay 100",
+			  zwave.basicV1.basicGet().format()])
+}
+
+def parse(String description) {
+	def result = null
+	def cmd = zwave.parse(description, [0x20: 1, 0x84: 1, 0x98: 1, 0x56: 1, 0x60: 3])
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	}
+	log.debug("'$description' parsed to $result")
+	return createEvent(result)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	[name: "switch", value: cmd.value ? "on" : "off"]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicSet cmd) {
+	[name: "switch", value: cmd.value ? "on" : "off"]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd) {
+	[name: "switch", value: cmd.value ? "on" : "off"]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x32: 3, 0x25: 1, 0x20: 1])
+	if (cmd.sourceEndPoint == 1) {
+		zwaveEvent(encapsulatedCommand)
+	} else { // sourceEndPoint == 2
+		childDevices[0]?.handleZWave(encapsulatedCommand)
+		[:]
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.crc16encapv1.Crc16Encap cmd) {
+	def versions = [0x31: 2, 0x30: 1, 0x84: 1, 0x9C: 1, 0x70: 2]
+	def version = versions[cmd.commandClass as Integer]
+	def ccObj = version ? zwave.commandClass(cmd.commandClass, version) : zwave.commandClass(cmd.commandClass)
+	def encapsulatedCommand = ccObj?.command(cmd.command)?.parse(cmd.data)
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	}
+	[:]
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	[descriptionText: "$device.displayName: $cmd", isStateChange: true]
+}
+
+def on() {
+	delayBetween([
+		zwave.switchBinaryV1.switchBinarySet(switchValue: 0xFF).format(),
```

Comment:
Does a switch binary set turn both lights off or just the first endpoint?

I would have expected you to need to encapsulate these commands.

Reply:
This is what configuration parameter does.

It decides which endpoints are controlled by binary switch set command sent from association group 1 (designed for controllers).

Configuration parameter 4 value:
0 - binary switch set commands from association group 1 will not control any endpoint. We would need to encapsulate to control each switch.
1 - binary switch set commands from association group 1 will control endpoint 1 (this is default setting)
2 - binary switch set commands from association group 1 will control endpoint 2
3 - binary switch set commands from association group 1 will control both endpoint 1 and endpoint 2

I decided to use default setting because it is default and it makes on() and off() commands for one endpoint faster.

I will add comments to explain this.

Reply:
Okay. Thanks.